packages/contracts/src/plugins/governance/majority-voting/token/TokenVotingSetup.sol

```diff
@@ -275,4 +233,25 @@ contract TokenVotingSetup is PluginSetup {
         );
         return success && data.length == 0x20;
     }
+
+    /// @notice Unsatisfiably determines if the token is an IVotes interface.
+    /// @dev Many tokens don't use ERC165 even though they still support IVotes.
+    function supportsIVotesInterface(address token) public view returns (bool) {
+        (bool success1, bytes memory data1) = token.staticcall(
```

Comment:
I believe this can be circumvented if you define a fallback function that simply returns `true`. 

Then you, as an attacker, would encode a malicious `initialize` function and you would have execution inside `prepareInstallation`.

Does that help you? I don't believe so tbh.

Reply:
The same can be true with ERC165 which is not the safe guard against malicious people, it's just used in good/legit people. With ERC165, you can have a fallback that returns true with all the things except if `0xffffffff` is passed in which case it returns false. The above check I have is the same kind of idea and doesn't protect it from malicious behaviour - you just can NOT and no one can avoid that malicious behaviour. That's why members have to review `applyinstallation` proposal in any case.

Reply:
fine